### PR TITLE
2d muvt

### PR DESCRIPTION
**Pull request by tcmoore3**
## Description

Add the ability to run the muVT updater in 2d systems. The only differences with the 3d case are 1) you generate a random element of SO(2) instead of SO(3) for the orientation of the test particle, and you only select random x and y components of the test particle position (and set z to the middle of the box in the z-direction).

## Motivation and Context

This enables muVT simulations for 2d systems. 

## How Has This Been Tested?

Added tests analogous to the 3d muVT tests, and have been using this for research.

## Change log

```
- Add muVT updater for 2d systems
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/hoomd-blue/blob/master/sphinx-doc/credits.rst).

### CODE REVIEW

File: hoomd/hpmc/Moves.h [hunk 1]
```diff
@@ -140,6 +140,15 @@ DEVICE inline quat<Scalar> generateRandomOrientation(RNG& rng)
 
     }
 
+//! Generate a random rotation about the z-axis
+template<class RNG>
+DEVICE inline quat<Scalar> generateRandomOrientation2D(RNG& rng)
+    {
+    Scalar u1 = hoomd::detail::generate_canonical<Scalar>(rng);
+    return quat<Scalar>(fast::cos(Scalar(2.0*M_PI)*(u1-Scalar(0.5))/Scalar(2.0)),
```
Comment (joaander) [inline hoomd/hpmc/Moves.h]: We can simplify the code:
```
Scalar theta = hoomd::detail::UniformDistribution<Scalar>(-M_PI, M_PI)(rng);
... fast::cos(theta) 
... fast::sin(theta)
```

File: hoomd/hpmc/UpdaterMuVT.h [hunk 2]
```diff
@@ -312,7 +312,8 @@ UpdaterMuVT<Shape>::UpdaterMuVT(std::shared_ptr<SystemDefinition> sysdef,
 
     if (m_sysdef->getNDimensions() == 2)
```
Comment (joaander) [inline hoomd/hpmc/UpdaterMuVT.h]: Remove the whole check. The constructing message has already been printed.

File: hoomd/hpmc/UpdaterMuVT.h [hunk 3]
```diff
@@ -643,15 +644,29 @@ void UpdaterMuVT<Shape>::update(unsigned int timestep)
...
                         {
                         // set particle orientation
+                        if (m_sysdef->getNDimensions() == 2)
+                        {
+                        shape_test.orientation = generateRandomOrientation2D(rng);
+                        }
+                        else
+                        {
                         shape_test.orientation = generateRandomOrientation(rng);
                         }
```
Comment (joaander) [inline hoomd/hpmc/UpdaterMuVT.h]: Indent the bodies of the conditional blocks.

File: hoomd/hpmc/Moves.h (hunk 1 above)
Comment (tcmoore3) [reply hoomd/hpmc/Moves.h]: 👍

File: hoomd/hpmc/UpdaterMuVT.h (hunk 2 above)
Comment (tcmoore3) [reply hoomd/hpmc/UpdaterMuVT.h]: 👍

File: hoomd/hpmc/UpdaterMuVT.h (hunk 3 above)
Comment (tcmoore3) [reply hoomd/hpmc/UpdaterMuVT.h]: 👍

**Comment by joaander** [comment]
@tcmoore3, 53fa0777350f7b merges this PR from _master_ up into _next_. Please verify that 2D muVT simulations function properly after the merge.

**Pull request merged**